src/algebraic_geometry/pullbacks.lean

```diff
@@ -0,0 +1,330 @@
+by { delta t', simp }
+
+@[simp, reassoc]
+lemma t'_fst_fst_snd (x y z : 𝒰.J) :
+  t' 𝒰 f g x y z ≫ pullback.fst ≫ pullback.fst ≫ pullback.snd =
+    pullback.fst ≫ pullback.fst ≫ pullback.snd :=
+by { delta t', simp }
+
+@[simp, reassoc]
+lemma t'_fst_snd (x y z : 𝒰.J) :
+  t' 𝒰 f g x y z ≫ pullback.fst ≫ pullback.snd = pullback.snd ≫ pullback.snd :=
+by { delta t', simp }
+
+@[simp, reassoc]
+lemma t'_snd_fst_fst (x y z : 𝒰.J) :
+  t' 𝒰 f g x y z ≫ pullback.snd ≫ pullback.fst ≫ pullback.fst = pullback.fst ≫ pullback.snd :=
+by { delta t', simp }
+
+@[simp, reassoc]
+lemma t'_snd_fst_snd (x y z : 𝒰.J) :
+  t' 𝒰 f g x y z ≫ pullback.snd ≫ pullback.fst ≫ pullback.snd =
+    pullback.fst ≫ pullback.fst ≫ pullback.snd :=
+by { delta t', simp }
+
+@[simp, reassoc]
+lemma t'_snd_snd (x y z : 𝒰.J) :
+  t' 𝒰 f g x y z ≫ pullback.snd ≫ pullback.snd = pullback.fst ≫ pullback.fst ≫ pullback.fst :=
+by { delta t', simp, }
+
+lemma cocycle_fst_fst_fst (x y z : 𝒰.J) :
+  t' 𝒰 f g x y z ≫ t' 𝒰 f g y z x ≫ t' 𝒰 f g z x y ≫ pullback.fst ≫ pullback.fst ≫
+  pullback.fst = pullback.fst ≫ pullback.fst ≫ pullback.fst :=
+by simp
+
+lemma cocycle_fst_fst_snd (x y z : 𝒰.J) :
+  t' 𝒰 f g x y z ≫ t' 𝒰 f g y z x ≫ t' 𝒰 f g z x y ≫ pullback.fst ≫ pullback.fst ≫
+  pullback.snd = pullback.fst ≫ pullback.fst ≫ pullback.snd :=
+by simp
+
+lemma cocycle_fst_snd (x y z : 𝒰.J) :
+  t' 𝒰 f g x y z ≫ t' 𝒰 f g y z x ≫ t' 𝒰 f g z x y ≫ pullback.fst ≫ pullback.snd =
+    pullback.fst ≫ pullback.snd :=
+by simp
+
+lemma cocycle_snd_fst_fst (x y z : 𝒰.J) :
+  t' 𝒰 f g x y z ≫ t' 𝒰 f g y z x ≫ t' 𝒰 f g z x y ≫ pullback.snd ≫ pullback.fst ≫
+  pullback.fst = pullback.snd ≫ pullback.fst ≫ pullback.fst :=
+by { rw ← cancel_mono (𝒰.map x), simp [pullback.condition_assoc, pullback.condition] }
+
+lemma cocycle_snd_fst_snd (x y z : 𝒰.J) :
+  t' 𝒰 f g x y z ≫ t' 𝒰 f g y z x ≫ t' 𝒰 f g z x y ≫ pullback.snd ≫ pullback.fst ≫
+  pullback.snd = pullback.snd ≫ pullback.fst ≫ pullback.snd :=
+by { simp [pullback.condition_assoc, pullback.condition] }
+
+lemma cocycle_snd_snd (x y z : 𝒰.J) :
+  t' 𝒰 f g x y z ≫ t' 𝒰 f g y z x ≫ t' 𝒰 f g z x y ≫ pullback.snd ≫ pullback.snd =
+    pullback.snd ≫ pullback.snd :=
+by simp
+
+-- by tidy should solve it but it timesout.
+lemma cocycle (x y z : 𝒰.J) :
+  t' 𝒰 f g x y z ≫ t' 𝒰 f g y z x ≫ t' 𝒰 f g z x y = 𝟙 _ :=
+begin
+  apply pullback.hom_ext; rw category.id_comp,
+  apply pullback.hom_ext,
+  apply pullback.hom_ext,
+  simp_rw category.assoc,
+  exact cocycle_fst_fst_fst 𝒰 f g x y z,
+  simp_rw category.assoc,
+  exact cocycle_fst_fst_snd 𝒰 f g x y z,
+  simp_rw category.assoc,
+  exact cocycle_fst_snd 𝒰 f g x y z,
+  apply pullback.hom_ext,
+  apply pullback.hom_ext,
+  simp_rw category.assoc,
+  exact cocycle_snd_fst_fst 𝒰 f g x y z,
+  simp_rw category.assoc,
+  exact cocycle_snd_fst_snd 𝒰 f g x y z,
+  simp_rw category.assoc,
+  exact cocycle_snd_snd 𝒰 f g x y z
+end
+
+/-- The glued fibered product `X ×[Z] Y` given `Uᵢ ×[Z] Y`. -/
```

Comment:
This sentence need is incomplete.

Reply:
I'm not sure about what you mean?

Reply:
Sorry, I meant that the sentence doesn't make sense standalone. (Ironically I managed to completely mangle up the sentence in which I tried to point this out :rofl:)